aws/cloudformation/iam.yml.erb

```
@@ -835,6 +843,438 @@ roles.each do |name, config| %>
+      # development: {
+      #   base_domain_name: 'marketing-sites.dev-code.org'
+      # },
+      test: {
+        base_domain_name: 'marketing-sites.test-code.org'
+      },
+      production: {
+        base_domain_name: 'code.org'
+      }
+    }
+    marketing_sites_environment_type_configuration.each do |environment_type, configuration|
+  %>
+  GitHubActions<%=environment_type.capitalize%>Role:
+    Type: AWS::IAM::Role
+    Properties:
+      RoleName: github-actions-cloudformation-<%=environment_type%>-role
+      Description: Role for GitHub Actions to create/update CloudFormation stacks.
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              Federated: !Sub "arn:aws:iam::${AWS::AccountId}:oidc-provider/token.actions.githubusercontent.com"
+            Action: sts:AssumeRoleWithWebIdentity
+            Condition:
+              StringEquals:
+                # See: https://docs.github.com/en/actions/security-for-github-actions/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services#configuring-the-role-and-trust-policy
+                token.actions.githubusercontent.com:aud: sts.amazonaws.com
+                token.actions.githubusercontent.com:sub: 'repo:code-dot-org/code-dot-org:environment:marketing-sites-<%=environment_type%>'
+      Policies:
+        - PolicyName: github-actions-cloudformation-policy
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Sid: CloudFormationPermissions
+                Effect: Allow
+                Action:
+                  - cloudformation:CreateStack
+                  - cloudformation:DescribeStacks
+                  - cloudformation:CreateChangeSet
+                  - cloudformation:DescribeChangeSet
+                  - cloudformation:DeleteChangeSet
+                  - cloudformation:ExecuteChangeSet
+                  - cloudformation:ListStackResources
+                Resource: '*'
+              - Sid: PassRoleToCloudFormation
+                Effect: Allow
+                Action:
+                  - iam:PassRole
+                Resource: !GetAtt CloudFormationServiceMarketingSites<%=environment_type.capitalize%>Role.Arn
+              # GitHub Actions needs to flush the content distribution network cache after invoking a successful Stack update.
+              - Sid: AllowCloudfrontInvalidation
+                Effect: Allow
+                Action:
+                  - cloudfront:CreateInvalidation
+                Resource: '*'
+                Condition:
+                  StringEquals:
+                    'aws:ResourceTag/environment-type': <%=environment_type%>
+              # GitHub Actions uploads Next.js assets (CSS, javascript, etc.) to the static assets bucket during each build.
+              - Sid: AllowS3StaticAssetsUpload
```

To differentiate from bucket permissions suggested above

```suggestion
              - Sid: AllowS3StaticAssetsUploadResources
```